### ISSUE TITLE
SEVERE: Error executing FreeMarker template

### ISSUE THREAD

**Issue by rcpoison** [stoicflame/enunciate#158]
Hi ^^
Again with 2.0.0-RC.1, disabling all modules but doc.

Unfortunately I haven't the slightest clue how to narrow this one down:

```
SEVERE: Error executing FreeMarker template
FreeMarker template error:
The following has evaluated to null or missing:
==> supertype.value.properties  [in template "jar:file:/home/poison/.m2/repository/com/webcohesion/enunciate/enunciate-docs/2.0.0-RC.1/enunciate-docs-2.0.0-RC.1.jar!/com/webcohesion/enunciate/modules/docs/docs.fmt" at line 929, column 42]

----
Tip: It's the step after the last dot that caused this error, not those before it.
----
Tip: If the failing expression is known to be legally refer to something that's null or missing, either specify a default value like myOptionalVar!myDefault, or use [#if myOptionalVar??]when-present[#else]when-missing[/#if]. (These only cover the last step of the expression; to cover the whole expression, use parenthesis: (myOptionalVar.foo)!myDefault, (myOptionalVar.foo)??
----

----
FTL stack trace ("~" means nesting-related):
        - Failed at: #if supertype.value?? && (supertype.v...  [in template "jar:file:/home/poison/.m2/repository/com/webcohesion/enunciate/enunciate-docs/2.0.0-RC.1/enunciate-docs-2.0.0-RC.1.jar!/com/webcohesion/enunciate/modules/docs/docs.fmt" in macro "processDataType" at line 929, column 15]
        ~ Reached through: #nested  [in template "jar:file:/home/poison/.m2/repository/com/webcohesion/enunciate/enunciate-docs/2.0.0-RC.1/enunciate-docs-2.0.0-RC.1.jar!/com/webcohesion/enunciate/modules/docs/docs.fmt" in macro "boilerplate" at line 119, column 9]
        ~ Reached through: @boilerplate title=(title + ": " + ty...  [in template "jar:file:/home/poison/.m2/repository/com/webcohesion/enunciate/enunciate-docs/2.0.0-RC.1/enunciate-docs-2.0.0-RC.1.jar!/com/webcohesion/enunciate/modules/docs/docs.fmt" in macro "processDataType" at line 848, column 5]
        - Reached through: @file name=(type.slug + ".html")  [in template "jar:file:/home/poison/.m2/repository/com/webcohesion/enunciate/enunciate-docs/2.0.0-RC.1/enunciate-docs-2.0.0-RC.1.jar!/com/webcohesion/enunciate/modules/docs/docs.fmt" in macro "processDataType" at line 847, column 3]
        - Reached through: @processDataType type=type  [in template "jar:file:/home/poison/.m2/repository/com/webcohesion/enunciate/enunciate-docs/2.0.0-RC.1/enunciate-docs-2.0.0-RC.1.jar!/com/webcohesion/enunciate/modules/docs/docs.fmt" in macro "processDataSyntax" at line 832, column 13]
        ~ Reached through: #nested  [in template "jar:file:/home/poison/.m2/repository/com/webcohesion/enunciate/enunciate-docs/2.0.0-RC.1/enunciate-docs-2.0.0-RC.1.jar!/com/webcohesion/enunciate/modules/docs/docs.fmt" in macro "boilerplate" at line 119, column 9]
        ~ Reached through: @boilerplate title=(title + ": " + sy...  [in template "jar:file:/home/poison/.m2/repository/com/webcohesion/enunciate/enunciate-docs/2.0.0-RC.1/enunciate-docs-2.0.0-RC.1.jar!/com/webcohesion/enunciate/modules/docs/docs.fmt" in macro "processDataSyntax" at line 813, column 5]
        - Reached through: @file name=(syntax.slug + ".html")  [in template "jar:file:/home/poison/.m2/repository/com/webcohesion/enunciate/enunciate-docs/2.0.0-RC.1/enunciate-docs-2.0.0-RC.1.jar!/com/webcohesion/enunciate/modules/docs/docs.fmt" in macro "processDataSyntax" at line 812, column 3]
        - Reached through: @processDataSyntax syntax=syntax  [in template "jar:file:/home/poison/.m2/repository/com/webcohesion/enunciate/enunciate-docs/2.0.0-RC.1/enunciate-docs-2.0.0-RC.1.jar!/com/webcohesion/enunciate/modules/docs/docs.fmt" at line 261, column 9]
        ~ Reached through: #nested  [in template "jar:file:/home/poison/.m2/repository/com/webcohesion/enunciate/enunciate-docs/2.0.0-RC.1/enunciate-docs-2.0.0-RC.1.jar!/com/webcohesion/enunciate/modules/docs/docs.fmt" in macro "boilerplate" at line 119, column 9]
        ~ Reached through: @boilerplate pagenav=pagenav  [in template "jar:file:/home/poison/.m2/repository/com/webcohesion/enunciate/enunciate-docs/2.0.0-RC.1/enunciate-docs-2.0.0-RC.1.jar!/com/webcohesion/enunciate/modules/docs/docs.fmt" at line 170, column 3]
        - Reached through: @file name=indexPageName  [in template "jar:file:/home/poison/.m2/repository/com/webcohesion/enunciate/enunciate-docs/2.0.0-RC.1/enunciate-docs-2.0.0-RC.1.jar!/com/webcohesion/enunciate/modules/docs/docs.fmt" at line 159, column 1]
----

Java stack trace (for programmers):
----
freemarker.core.InvalidReferenceException: [... Exception message was already printed; see it above ...]
        at freemarker.core.InvalidReferenceException.getInstance(InvalidReferenceException.java:116)
        at freemarker.core.UnexpectedTypeException.newDesciptionBuilder(UnexpectedTypeException.java:60)
        at freemarker.core.UnexpectedTypeException.<init>(UnexpectedTypeException.java:40)
        at freemarker.core.BuiltInsForMultipleTypes$sizeBI._eval(BuiltInsForMultipleTypes.java:448)
        at freemarker.core.Expression.eval(Expression.java:78)
        at freemarker.core.EvalUtil.compare(EvalUtil.java:110)
        at freemarker.core.ComparisonExpression.evalToBoolean(ComparisonExpression.java:64)
        at freemarker.core.ParentheticalExpression.evalToBoolean(ParentheticalExpression.java:31)
        at freemarker.core.AndExpression.evalToBoolean(AndExpression.java:32)
        at freemarker.core.ConditionalBlock.accept(ConditionalBlock.java:46)
        at freemarker.core.Environment.visitByHiddingParent(Environment.java:333)
        at freemarker.core.IteratorBlock$Context.runLoop(IteratorBlock.java:159)
        at freemarker.core.Environment.visitIteratorBlock(Environment.java:559)
        at freemarker.core.IteratorBlock.accept(IteratorBlock.java:67)
        at freemarker.core.Environment.visitByHiddingParent(Environment.java:333)
        at freemarker.core.ConditionalBlock.accept(ConditionalBlock.java:48)
        at freemarker.core.Environment.visit(Environment.java:312)
        at freemarker.core.MixedContent.accept(MixedContent.java:62)
        at freemarker.core.Environment.visitByHiddingParent(Environment.java:333)
        at freemarker.core.ConditionalBlock.accept(ConditionalBlock.java:48)
        at freemarker.core.Environment.visit(Environment.java:312)
        at freemarker.core.MixedContent.accept(MixedContent.java:62)
        at freemarker.core.Environment.visit(Environment.java:312)
        at freemarker.core.Environment.invokeNestedContent(Environment.java:535)
        at freemarker.core.BodyInstruction.accept(BodyInstruction.java:56)
        at freemarker.core.Environment.visit(Environment.java:312)
        at freemarker.core.MixedContent.accept(MixedContent.java:62)
        at freemarker.core.Environment.visit(Environment.java:312)
        at freemarker.core.Macro$Context.runMacro(Macro.java:178)
        at freemarker.core.Environment.invoke(Environment.java:700)
        at freemarker.core.UnifiedCall.accept(UnifiedCall.java:84)
        at freemarker.core.Environment.visit(Environment.java:312)
        at freemarker.core.Environment$NestedElementTemplateDirectiveBody.render(Environment.java:2255)
        at com.webcohesion.enunciate.util.freemarker.FileDirective.execute(FileDirective.java:59)
        at freemarker.core.Environment.visit(Environment.java:377)
        at freemarker.core.UnifiedCall.accept(UnifiedCall.java:104)
        at freemarker.core.Environment.visit(Environment.java:312)
        at freemarker.core.MixedContent.accept(MixedContent.java:62)
        at freemarker.core.Environment.visit(Environment.java:312)
        at freemarker.core.Macro$Context.runMacro(Macro.java:178)
        at freemarker.core.Environment.invoke(Environment.java:700)
        at freemarker.core.UnifiedCall.accept(UnifiedCall.java:84)
        at freemarker.core.Environment.visit(Environment.java:312)
        at freemarker.core.MixedContent.accept(MixedContent.java:62)
        at freemarker.core.Environment.visitByHiddingParent(Environment.java:333)
        at freemarker.core.IteratorBlock$Context.runLoop(IteratorBlock.java:159)
        at freemarker.core.Environment.visitIteratorBlock(Environment.java:559)
        at freemarker.core.IteratorBlock.accept(IteratorBlock.java:67)
        at freemarker.core.Environment.visit(Environment.java:312)
        at freemarker.core.MixedContent.accept(MixedContent.java:62)
        at freemarker.core.Environment.visitByHiddingParent(Environment.java:333)
        at freemarker.core.ConditionalBlock.accept(ConditionalBlock.java:48)
        at freemarker.core.Environment.visitByHiddingParent(Environment.java:333)
        at freemarker.core.IteratorBlock$Context.runLoop(IteratorBlock.java:159)
        at freemarker.core.Environment.visitIteratorBlock(Environment.java:559)
        at freemarker.core.IteratorBlock.accept(IteratorBlock.java:67)
        at freemarker.core.Environment.visit(Environment.java:312)
        at freemarker.core.MixedContent.accept(MixedContent.java:62)
        at freemarker.core.Environment.visit(Environment.java:312)
        at freemarker.core.Environment.invokeNestedContent(Environment.java:535)
        at freemarker.core.BodyInstruction.accept(BodyInstruction.java:56)
        at freemarker.core.Environment.visit(Environment.java:312)
        at freemarker.core.MixedContent.accept(MixedContent.java:62)
        at freemarker.core.Environment.visit(Environment.java:312)
        at freemarker.core.Macro$Context.runMacro(Macro.java:178)
        at freemarker.core.Environment.invoke(Environment.java:700)
        at freemarker.core.UnifiedCall.accept(UnifiedCall.java:84)
        at freemarker.core.Environment.visit(Environment.java:312)
        at freemarker.core.Environment$NestedElementTemplateDirectiveBody.render(Environment.java:2255)
        at com.webcohesion.enunciate.util.freemarker.FileDirective.execute(FileDirective.java:59)
        at freemarker.core.Environment.visit(Environment.java:377)
        at freemarker.core.UnifiedCall.accept(UnifiedCall.java:104)
        at freemarker.core.Environment.visit(Environment.java:312)
        at freemarker.core.MixedContent.accept(MixedContent.java:62)
        at freemarker.core.Environment.visit(Environment.java:312)
        at freemarker.core.Macro$Context.runMacro(Macro.java:178)
        at freemarker.core.Environment.invoke(Environment.java:700)
        at freemarker.core.UnifiedCall.accept(UnifiedCall.java:84)
        at freemarker.core.Environment.visit(Environment.java:312)
        at freemarker.core.MixedContent.accept(MixedContent.java:62)
        at freemarker.core.Environment.visitByHiddingParent(Environment.java:333)
        at freemarker.core.IteratorBlock$Context.runLoop(IteratorBlock.java:159)
        at freemarker.core.Environment.visitIteratorBlock(Environment.java:559)
        at freemarker.core.IteratorBlock.accept(IteratorBlock.java:67)
        at freemarker.core.Environment.visit(Environment.java:312)
        at freemarker.core.MixedContent.accept(MixedContent.java:62)
        at freemarker.core.Environment.visitByHiddingParent(Environment.java:333)
        at freemarker.core.ConditionalBlock.accept(ConditionalBlock.java:48)
        at freemarker.core.Environment.visit(Environment.java:312)
        at freemarker.core.MixedContent.accept(MixedContent.java:62)
        at freemarker.core.Environment.visit(Environment.java:312)
        at freemarker.core.Environment.invokeNestedContent(Environment.java:535)
        at freemarker.core.BodyInstruction.accept(BodyInstruction.java:56)
        at freemarker.core.Environment.visit(Environment.java:312)
        at freemarker.core.MixedContent.accept(MixedContent.java:62)
        at freemarker.core.Environment.visit(Environment.java:312)
        at freemarker.core.Macro$Context.runMacro(Macro.java:178)
        at freemarker.core.Environment.invoke(Environment.java:700)
        at freemarker.core.UnifiedCall.accept(UnifiedCall.java:84)
        at freemarker.core.Environment.visit(Environment.java:312)
        at freemarker.core.MixedContent.accept(MixedContent.java:62)
        at freemarker.core.Environment.visit(Environment.java:312)
        at freemarker.core.Environment$NestedElementTemplateDirectiveBody.render(Environment.java:2255)
        at com.webcohesion.enunciate.util.freemarker.FileDirective.execute(FileDirective.java:59)
        at freemarker.core.Environment.visit(Environment.java:377)
        at freemarker.core.UnifiedCall.accept(UnifiedCall.java:104)
        at freemarker.core.Environment.visit(Environment.java:312)
        at freemarker.core.MixedContent.accept(MixedContent.java:62)
        at freemarker.core.Environment.visit(Environment.java:312)
        at freemarker.core.Environment.process(Environment.java:290)
        at freemarker.template.Template.process(Template.java:312)
        at com.webcohesion.enunciate.modules.docs.DocsModule.processTemplate(DocsModule.java:350)
        at com.webcohesion.enunciate.modules.docs.DocsModule.call(DocsModule.java:302)
        at com.webcohesion.enunciate.io.InvokeEnunciateModule.onNext(InvokeEnunciateModule.java:31)
        at com.webcohesion.enunciate.io.InvokeEnunciateModule.onNext(InvokeEnunciateModule.java:10)
        at rx.internal.operators.OperatorDoOnEach$1.onNext(OperatorDoOnEach.java:79)
        at rx.internal.operators.OperatorSingle$ParentSubscriber.onCompleted(OperatorSingle.java:109)
        at rx.internal.operators.NotificationLite.accept(NotificationLite.java:140)
        at rx.internal.operators.TakeLastQueueProducer.emit(TakeLastQueueProducer.java:100)
        at rx.internal.operators.TakeLastQueueProducer.startEmitting(TakeLastQueueProducer.java:45)
        at rx.internal.operators.OperatorTakeLast$1.onCompleted(OperatorTakeLast.java:59)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
        at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:658)
        at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
        at rx.subjects.SubjectSubscriptionManager$SubjectObserver.onCompleted(SubjectSubscriptionManager.java:232)
        at rx.subjects.ReplaySubject.onCompleted(ReplaySubject.java:404)
        at rx.Observable$35.onCompleted(Observable.java:7458)
        at rx.observers.SafeSubscriber.onCompleted(SafeSubscriber.java:81)
        at rx.internal.operators.OperatorDoOnEach$1.onCompleted(OperatorDoOnEach.java:53)
        at rx.internal.operators.OperatorSingle$ParentSubscriber.onCompleted(OperatorSingle.java:110)
        at rx.internal.operators.NotificationLite.accept(NotificationLite.java:140)
        at rx.internal.operators.TakeLastQueueProducer.emit(TakeLastQueueProducer.java:100)
        at rx.internal.operators.TakeLastQueueProducer.startEmitting(TakeLastQueueProducer.java:45)
        at rx.internal.operators.OperatorTakeLast$1.onCompleted(OperatorTakeLast.java:59)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
        at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:658)
        at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
        at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:673)
        at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
        at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:673)
        at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
        at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:673)
        at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
        at rx.internal.util.ScalarSynchronousObservable$ScalarSynchronousAction.call(ScalarSynchronousObservable.java:117)
        at rx.schedulers.ExecutorScheduler$ExecutorAction.run(ExecutorScheduler.java:173)
        at rx.schedulers.ExecutorScheduler$ExecutorSchedulerWorker.run(ExecutorScheduler.java:99)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
        at java.lang.Thread.run(Thread.java:745)
```

**Comment by stoicflame** [comment]
Fixed at a847798. I've deployed 2.0.0-SNAPSHOT; you're welcome to try it out if you want to use the [OSSRH snapshot repository](https://oss.sonatype.org/content/repositories/snapshots/).

**Comment by rcpoison** [comment]
I don't know how to thank you :) I can finally build the project with java8 ^^
